src/common/HeartbeatMap.cc

```diff
@@ -64,19 +64,19 @@ void HeartbeatMap::remove_worker(const heartbeat_handle_d *h)
   delete h;
 }
 
-bool HeartbeatMap::_check(const heartbeat_handle_d *h, const char *who, time_t now)
+bool HeartbeatMap::_check(const heartbeat_handle_d *h, const char *who, ceph::coarse_mono_clock::time_point &now)
 {
   bool healthy = true;
-  time_t was;
+  unsigned was;
 
   was = h->timeout;
-  if (was && was < now) {
+  if (was && was < std::chrono::time_point_cast<std::chrono::seconds>(now).time_since_epoch().count()) {
```

Comment:
would be good to use typdef for 'std::chrono::time_point_cast<std::chrono::seconds>' instead repeating again & again